src/api/cli/integration/integration_test.go

```diff
@@ -1341,3 +1357,236 @@ func TestLiveWalletDir(t *testing.T) {
+		return
+	}
+
+	w := prepareAndCheckWallet(t, 4)
+
+	tt := []struct {
+		name    string
+		args    func() []string
+		checkTx func(t *testing.T, txid string)
+	}{
+		{
+			"send with -a(from address) flag",
+			func() []string {
+				// Set the third address as the receiver address
+				return []string{"send", "-a", w.Entries[0].Address.String(), w.Entries[1].Address.String(), "0.5"}
+			},
+			func(t *testing.T, txid string) {},
+		},
+		{
+			"send to multiple address",
+			func() []string {
+				addrCoins := []struct {
+					Addr  string `json:"addr"`
+					Coins string `json:"coins"`
+				}{
+					{
+						w.Entries[1].Address.String(),
+						"0.5",
+					},
+					{
+						w.Entries[2].Address.String(),
+						"0.5",
+					},
+				}
+
+				v, err := json.Marshal(addrCoins)
+				require.NoError(t, err)
+
+				return []string{"send", "-m", string(v)}
+			},
+			func(t *testing.T, txid string) {
+				tx := getTransaction(t, txid)
+				// get the total input hours the transaction has.
+				_, hours := getUxoutCoinsAndHours(t, tx.Transaction.Transaction.In)
+				// Only when hours >= 4, the two recipient addresses could all receive 1 coinhours
+				if hours >= 4 {
+					// Confirms the second address has 0.5 coin and 1 coinhour
+					checkCoinsAndCoinhours(t, tx, w.Entries[1].Address.String(), "0.500000", 1)
+					// Confirms the third address has 1 coin and 1 coinhour
+					checkCoinsAndCoinhours(t, tx, w.Entries[2].Address.String(), "0.500000", 1)
+				}
+
+				// If the total input coin hours is <= 1, the two recipient addresses should receive no coinhours.
+				if hours <= 1 {
+					// Confirms the second address has 0.5 coin and 1 coinhour
+					checkCoinsAndCoinhours(t, tx, w.Entries[1].Address.String(), "0.500000", 0)
+					// Confirms the third address has 1 coin and 1 coinhour
+					checkCoinsAndCoinhours(t, tx, w.Entries[2].Address.String(), "0.500000", 0)
+				}
+			},
+		},
+		{
+			"send with -c(change address) flag",
+			func() []string {
+				// Set the third address as change address
+				return []string{"send", "-c", w.Entries[2].Address.String(), w.Entries[0].Address.String(), "1"}
+			},
+			func(t *testing.T, txid string) {},
+		},
+		{
+			"send to one address",
+			func() []string {
+				return []string{"send", w.Entries[0].Address.String(), "2"}
+			},
+			func(t *testing.T, txid string) {},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			output, err := exec.Command(binaryPath, tc.args()...).CombinedOutput()
+			if err != nil {
+				fmt.Println("result:", output)
+			}
+			require.NoError(t, err)
+			output = bytes.TrimRight(output, "\n")
+
+			// output: "txid:$txid_string"
+			// split the output to get txid value
+			v := bytes.Split(output, []byte(":"))
+			require.Len(t, v, 2)
+			txid := string(v[1])
+			fmt.Println("txid:", txid)
+			_, err = cipher.SHA256FromHex(txid)
+			require.NoError(t, err)
+
+			// Wait untill transaction is confirmed.
+			for {
+				time.Sleep(time.Second)
```

Comment:
Should use a for select loop with a ticker for checking confirmation and a ticker for timeout (maybe 30 seconds timeout?)